### ISSUE TITLE
Antialiasing issues

### ISSUE THREAD

**Issue by kkizlaitis** [xvrh/lottie-flutter#82]
First of all, thank you for this package!

I have downloaded a very simple animation (just an inflating red circle): [Recording](https://assets7.lottiefiles.com/packages/lf20_lcwZD8.json)

Here is a screenshot of how it looks on an iPhone X:
![Screenshot 2020-07-08 at 17 21 23](https://user-images.githubusercontent.com/29122296/86930429-aeeba280-c13f-11ea-93a3-f657bd9785fb.jpeg)

However, when I render it, it appears pixelated around the edges. Maybe it is because I show it very small? I specify `width: 30` and `height: 30`.

Would love to hear what could be done to resolve the sharp edges of the circle.

**Comment by 152dyn** [comment]
I had the same problem with my iPhone 11Pro, which worked well on Android

**Comment by kkizlaitis** [comment]
I also saw that it doesn't matter what the size of the animation is. My device was iPhone X (not a simulator).

**Comment by 152dyn** [comment]
Yes, it has nothing to do with size. Have you found a solution later

**Comment by kkizlaitis** [comment]
Not yet, for now I left it like that... Input from @xvrh would be greatly appreciated 😊

**Comment by 152dyn** [comment]
I covered the serrate with a line of the same color to solve this problem on the iPhone, I hope it helps you.

**Comment by flylxc** [comment]
I encountered the same aliasing problem with iPhone 11 devices.

**Comment by yuhaocan** [comment]
I have same problem.but it cant find in Android.
I see circle draw use path to do will have Antialiasing.

**Comment by yuhaocan** [comment]
In fill_content.dart I modify draw func,if is iOS and hasCircle use drawCircle instead of drawPath.
```
// canvas.drawPath(_path, _paint);
if(Platform.isIOS){
  if(hasCircle){
    var rect = _path.getBounds();
    canvas.drawCircle(rect.center,rect.width / 2,_paint);
  }else{
    canvas.drawPath(_path, _paint);
  }
}else{
  canvas.drawPath(_path, _paint);
}
```

**Comment by wingsico** [comment]
> In fill_content.dart I modify draw func,if is iOS and hasCircle use drawCircle instead of drawPath.
> 
> ```
> // canvas.drawPath(_path, _paint);
> if(Platform.isIOS){
>   if(hasCircle){
>     var rect = _path.getBounds();
>     canvas.drawCircle(rect.center,rect.width / 2,_paint);
>   }else{
>     canvas.drawPath(_path, _paint);
>   }
> }else{
>   canvas.drawPath(_path, _paint);
> }
> ```
> 
> It is not very well solution,but can solve my problem.Hope it can help you.

How do I know about `hasCircle`?

**Comment by yuhaocan** [comment]
> > In fill_content.dart I modify draw func,if is iOS and hasCircle use drawCircle instead of drawPath.
> > ```
> > // canvas.drawPath(_path, _paint);
> > if(Platform.isIOS){
> >   if(hasCircle){
> >     var rect = _path.getBounds();
> >     canvas.drawCircle(rect.center,rect.width / 2,_paint);
> >   }else{
> >     canvas.drawPath(_path, _paint);
> >   }
> > }else{
> >   canvas.drawPath(_path, _paint);
> > }
> > ```
> > 
> > 
> > It is not very well solution,but can solve my problem.Hope it can help you.
> 
> How do I know about `hasCircle`?

Find CircleShape and you can see EllipseContent.
If fill_content has  EllipseContent mean have circle.

**Comment by eggfly** [comment]
`final hasCircle = _path.any((path) {
  return path is EllipseContent;
})
`

This code may work but it needs some more tests
